Governance/tech/conventions.md

```diff
@@ -1,147 +1,295 @@
+If `_FillValue` is defined then it should be scalar and of the same type as the variable. If the variable is packed using scale_factor and add_offset attributes, the _FillValue attribute should have the data type of the packed data.<br>
+If not defined the deafult fill value for the type of the variable is used. However, use of the default fill value for data type byte is not recommended.<br>
+Note that changing the value of this attibute, won't changed previously 'filled' data automatically.<br>
+If `valid_range` is specified `_FillValue` should be outside of this range.
+
+**missing-value**<br>
+The `missing_value` is not treated in any special way by the netCDF library, but it may be used by specific applications. The missing_value attribute can be a scalar or vector containing values indicating missing data. These values should all be outside the valid range.<br>
+When scale_factor and add_offset are used, the value(s) of the missing_value attribute should be specified in relation to the packed data, so that missing values can be detected before the scale_factor and add_offset are applied.<br>
+If both missing_value and _FillValue are used, they should have the same value.
+
+**actual_range**<br>
+The `actual_range` defines a two-element vector for numeric variables, composed of the exact minimum and the maximum data values and both must be within the valid_range if specified.<br>
+If the variable is packed, the elements of the actual_range should be defined based on the unpacked data, including the type.<br>
+If the data is all missing or invalid, the actual_range attribute cannot be used.
+
+**valid_range, valid_min, valid_max**<br>
+The `valid_range` attribute is mutually exclusive with `valid_min` and `valid_max` attributes. If none of them is defined software applications will use `_Fill_value` and the variable type to try to determine a valid range.
 ```
 
-```{dropdown} ancillary data
+```{dropdown} **ancillary data**
 CF -
 ```
 
-```{dropdown} coverage_content_type
-ACDD - highly recommended 
+```{dropdown} **coverage_content_type**
+**ACDD - highly recommended** 
 ```
 
-Important link:
-https://cfconventions.org/Data/cf-documents/requirements-recommendations/conformance-1.8.html
-
-exceptions for boundary and climatology variables
+````{warning}
+In xarray variable attributes are kept across operations depending on the value of the `keep_attrs` argument. Its default behaviour is to keep them in unambiguous circumstances. If set to True it will always keep the attributes, if set to False it will discard them.<br>
+It is important to be aware of this, as particularly for coordinates, units and cell_methods are easily changed by calculations and often inherited attributes become meaningless or worst can cause issues if not updated. The same can also happen with other softwares.
+````
```

Comment:
I would be inclined, perhaps around about here, to provide info on how people can check their own files using online tools (CF-checker) or on NCI (cchecker.py), so that this page acts as a reinforcement to the output of that tool, but also people don't think they have to just read about how to meet standards, they can programmatically check files themselves?

Reply:
Good idea, in the concepts/cf-conventions we refer to the CMS wiki, this page: http://climate-cms.wikis.unsw.edu.au/CF_checker
As the wiki will disappear I could just add this as tech/cf-checker.md and then we can refer to it somewhere in this page

Reply:
Let me know if this sounds reasonable so I can add the new page and update the references, before merging

Reply:
Yes I think that's a good idea, if you want this book to succeed the wiki then it makes sense to move/copy useful information in it to here - this book is aimed at the Australian community so I think it's okay to have NCI-specific content. That page was a very useful reference for me so likely will be for others.

Reply:
Done, I've added a warning/reference at the top of this page and the cf-checker as a separate file that will show as the next page.